GitUI/UserControls/RevisionGridClasses/FormQuickGitRefSelector.cs

```diff
@@ -0,0 +1,128 @@
+        private readonly TranslationString _actionDelete = new TranslationString("Delete");
+        private readonly TranslationString _tag = new TranslationString("tag");
+
+
+        public FormQuickGitRefSelector()
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+
+        /// <summary>
+        /// Gets the ref selected by the user.
+        /// </summary>
+        public IGitRef SelectedRef => (lbxRefs.SelectedItem as DisplyGitRef)?.Item;
+
+
+        public void Init(Action action, IGitRef[] refs)
+        {
+            switch (action)
+            {
+                case Action.Delete:
+                    {
+                        btnAction.Text = _actionDelete.Text;
+                    }
+                    break;
+                case Action.Rename:
+                    {
+                        btnAction.Text = _actionRename.Text;
+                    }
+                    break;
+            }
+
+            lbxRefs.Items.Clear();
+            if (refs == null || refs.Length < 1)
+            {
+                DialogResult = DialogResult.Cancel;
+                Close();
+                return;
+            }
+
+            // constrain the listbox min width, otherwise the button may not have enough space
+            var longestItemWidth = 150f;
+            try
+            {
+                lbxRefs.BeginUpdate();
+
+                using (Graphics graphics = lbxRefs.CreateGraphics())
+                {
+                    foreach (var gitRef in refs.OrderBy(r => r.IsTag).ThenBy(r => r.Name))
+                    {
+                        var suffix = gitRef.IsTag ? $" ({_tag.Text})" : string.Empty;
+                        var item = new DisplyGitRef
+                        {
+                            Label = $"{gitRef.Name}{suffix}",
+                            Item = gitRef
+                        };
+                        lbxRefs.Items.Add(item);
+
+                        // assume that the branch names or tags are never longer than 100 symbols long
+                        // if they are (sanity!) - don't resize past 100 symbols
+                        var label = item.Label.Length > 100 ? item.Label.Substring(0, 100) : item.Label;
+                        longestItemWidth = Math.Max(longestItemWidth, graphics.MeasureString(label, lbxRefs.Font).Width);
+                    }
+                }
+            }
+            finally
+            {
+                // what is this magic number "8.5"?
+                // we don't want to show more than 8 items without scroll, this is to reduce the risk of not fitting on user's screen.
+                // so we want 8 items and a little bit more to account for listbox's border.
+                // the listbox will automatically recalculate and set its preferred height
+                lbxRefs.Height = Math.Min(lbxRefs.PreferredHeight, (int)(8.5 * lbxRefs.ItemHeight));
```

Comment:
I will turn 8 into a const, something I should have done initially.
Personally I don't see any point making it a configurable value, but I am happy to discuss.

With regards to calculation is seems a bit unintuitive with the listbox calculating its heights automatically. I tried few different approaches, and this seems to produce the most accurate result on the screen.